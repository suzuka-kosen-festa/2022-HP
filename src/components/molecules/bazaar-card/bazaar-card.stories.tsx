import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { BazaarCard } from ".";

type T = typeof BazaarCard;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  bazaar: {
    id: 1,
    name: "A.作品展示",
    descriptions: "美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！",
    image: "https://panproduct.com/blog/wp-content/uploads/2021/07/32.png",
    prices: [{ id: 1, price: "無料", bazaarId: 1 }],
    group: "美術部",
    group_type: "recreation",
  },
};

export default {
  args,
  argTypes: {
    bazaar: {
      description: "Bazaar data",
      control: {
        type: "object",
      },
    },
  },
  component: BazaarCard,
} as Meta;

export const Default: Story = {};
