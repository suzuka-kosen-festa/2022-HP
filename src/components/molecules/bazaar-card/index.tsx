import type { FC } from "react";
import tw, {css} from "twin.macro";
import type { BazaarCardProperties, ChipProperties } from "./types/model";

const ChipBox = tw.div`inline-flex items-start bg-primary-purple px-4 py-1 rounded-2xl`

const ChipText = tw.p`font-zen text-white font-bold text-xs sm:text-base select-none`

const BazaarCardBox = tw.li`inline-flex flex-col xs:w-75 border border-border-primary rounded-2xl overflow-hidden`

const BazaarCardImage = tw.img`h-full`

const BazaarCardContent = tw.div`inline-flex flex-col items-start px-2 py-2 space-y-2 sm:(py-4 space-y-4)`

const BazaarCardHeading = tw.h2`font-zen font-bold text-text text-base sm:text-2xl select-none`

const BazaarCardDescription = tw.p`font-zen font-bold text-gray-700 text-xs sm:text-base select-none`

const BazaarCardPrices = tw.ul`flex space-x-2 sm:space-x-3`

const BazaarCardPrice = tw.p`font-zen font-bold text-text text-xs sm:text-base select-none`

const Chip: FC<ChipProperties> = ({group}) => (
  <ChipBox>
    <ChipText>{group}</ChipText>
  </ChipBox>
)

const BazaarCard: FC<BazaarCardProperties> = ({ name, description, image, prices, group }) => (
  <BazaarCardBox>
    <BazaarCardImage css={css`aspect-ratio: 3 / 2;`} src={image} alt="バザーに関するイメージ画像" width={300} height={200} />
    <BazaarCardContent>
      <Chip group={group} />
      <BazaarCardHeading>{name}</BazaarCardHeading>
      <BazaarCardDescription css={css`width: fit-content;`}>{description}</BazaarCardDescription>
      <BazaarCardPrices>
        {prices.map((price) => (
          <li key={price}>
            <BazaarCardPrice>{price}</BazaarCardPrice>
          </li>
        ))}
      </BazaarCardPrices>
    </BazaarCardContent>
  </BazaarCardBox>
);

export { BazaarCard };
