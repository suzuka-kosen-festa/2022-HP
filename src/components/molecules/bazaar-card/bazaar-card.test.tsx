import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./bazaar-card.stories";

const { Default } = composeStories(stories);

describe("(components) molecules/bazaar-card", () => {
  test("to be molecules", () => {
    const { container } = render(<Default />);
    expect(container).toBeMolecule();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
