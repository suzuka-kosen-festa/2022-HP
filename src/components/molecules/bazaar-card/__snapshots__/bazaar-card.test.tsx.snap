// Vitest Snapshot v1

exports[`(components) molecules/bazaar-card > take snap shot 1`] = `
<div>
  <li
    class="css-4chdo6"
  >
    <img
      alt="バザーに関するイメージ画像"
      class="css-61q9gc-BazaarCard"
      height="300"
      src="https://panproduct.com/blog/wp-content/uploads/2021/07/32.png"
      width="450"
    />
    <div
      class="css-z0erpm"
    >
      <div
        class="css-1q8fkzn"
      >
        <p
          class="css-jprqak"
        >
          美術部
        </p>
      </div>
      <h2
        class="css-hh2gwj"
      >
        A.作品展示
      </h2>
      <p
        class="css-vaqku0-BazaarCard"
      >
        美術部員が描き溜めてきた作品たちを展示します。ぜひ見に来てね！
      </p>
      <ul
        class="css-luxo22"
      >
        <li>
          <p
            class="css-4gn9jk"
          >
            無料
          </p>
        </li>
      </ul>
    </div>
  </li>
</div>
`;
